src/blob/index/core.rs

```diff
@@ -244,9 +244,28 @@ where
         }
     }
 
-    fn mark_all_as_deleted_in_memory(headers: &mut InMemoryIndex<K>, key: &K) -> Option<u64> {
+    fn push_deletion_header_in_memory(
```

Comment:
You should also update `filters` to make the deletion visible (this will be required for the next PR)